File: frontend/src/service/amenityService.ts
import api from "../config/axios";
import type { ApiResponse } from "../types/ApiResponse";

export interface AmenityResponse {
  amenityId: number;
  amenityName: string;
  iconKey: string;
}

/*

AmenityResponse
export interface CityResponse {
  id: number;
  name: string;
}

*/
class AmenityService {
  async getAllAmenities(): Promise<ApiResponse<AmenityResponse[]>> {
    const res = await api.get("/amenities");
    return res.data;
  }

  async getAmenityById(id: number): Promise<ApiResponse<AmenityResponse>> {
    const res = await api.get(`/amenities/${id}`);
    return res.data;
  }
}

export default new AmenityService();

File: mobile/lib/services/amenity_service.dart
import '../models/amenity.dart';
import '../config/api_client.dart';
class AmenityService {
  Future<List<AmenityResponse>> getAllAmenities() async {
    final res = await apiClient.get('/amenities');

    final List data = res.data['result'] ?? [];

    return data.map((e) => AmenityResponse.fromJson(e)).toList();
  }

  Future<AmenityResponse> getAmenityById(int id) async {
    final res = await apiClient.get('/amenities/$id');

    return AmenityResponse.fromJson(res.data['result']);
  }
}

final amenityService = AmenityService();